tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -266,11 +310,16 @@ policy_refresh_cagg_add(PG_FUNCTION_ARGS)
 	namestrcpy(&owner, GetUserNameFromId(owner_id, false));
 
 	if (!start_isnull)
-		check_valid_interval(dim_type, start_interval_type, "start_interval");
+		check_valid_interval_type(dim_type, start_interval_type, "start_interval");
 	if (!end_isnull)
-		check_valid_interval(dim_type, end_interval_type, "end_interval");
+		check_valid_interval_type(dim_type, end_interval_type, "end_interval");
+
+	if (!(start_isnull || end_isnull))
+	{
+		Assert(start_interval_type == end_interval_type);
```

Comment:
I am not really sure that is correct assumption, maybe we should error instead? This will likely fail if we pass start_interval::smallint and end_interval::int as arguments, unless I am mistaken

Reply:
yes, it should. We check that the interval_type matches the dim_type just before this (line 313). Even with the PR change for automatically discovering the types , the interval types should be the same (and match the dimension type) before we compare their values.